### PR TITLE
Add support for React > v17

### PR DESCRIPTION
**Pull request by mw-sezzle**

### CODE REVIEW

**Comment by mw-sezzle** [comment]
@alexpanov I've tested this as functional by forcing resolutions via `npm install --force` alongside react v18.

File: .npmignore [hunk 1]
```diff
@@ -7,3 +7,11 @@ docs
...
 .scripts
 .storybook
+.yarn
+*.tgz
+*.log
+.github
+.eslintignore
+.eslintrc.js
+.yarnrc.yml
+public
```
Comment (alexpanov) [inline .npmignore]: Can you please add a newline here? @mw-sezzle

File: package.json [hunk 2]
```diff
@@ -1,12 +1,12 @@
 {
-  "name": "storybook-addon-material-ui5",
-  "version": "1.0.0",
+  "name": "storybook-addon-mui",
```
Comment (alexpanov) [inline package.json]: Do you insist we change the package name? @mw-sezzle

File: README.md [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
 #  Storybook Addon Material-UI
 
+## A big thanks to Alex Panov and all the devs previously who contributed to this project. This is primarily a republished package of `storybook-addon-material-ui-5`
```
Comment (alexpanov) [inline README.md]: Well, thank you, but that on my own page would look like ass kiss :D
Can you add yourself in there @mw-sezzle? =)

**Review by alexpanov — CHANGES_REQUESTED** [review]
Awesome work, @mw-sezzle! Thank you.
Can you tell me a bit more about your testing strategy? Just curious, how did you go about testing it? We're migrating a big project to a newer React version, so your input might be also useful there.
Cheers.

File: package.json [hunk 4]
```diff
@@ -99,7 +99,7 @@
     "@storybook/addons": "^6.3.12",
     "@storybook/react": "^6.3.12",
     "prop-types": "^15.5.8",
-    "react": "^17.0.0",
-    "react-dom": "^17.0.0"
+    "react": ">=17.0.0",
```
Comment (alexpanov) [inline package.json:102]: 👍

**Comment by alexpanov** [comment]
@mw-sezzle do you use the [current package](https://www.npmjs.com/package/storybook-addon-material-ui5) from npm somewhere in your projects?

File: README.md (hunk 3 above)
Comment (mw-sezzle) [reply README.md]: oh whoops, I wasn't sure how closely you maintained this project. I may have jumped the gun and republished. I should have opened a new branch for that work. Let me redo some of this.

File: package.json (hunk 2 above)
Comment (mw-sezzle) [reply package.json]: no, let me redo some of this.

File: .npmignore (hunk 1 above)
Comment (mw-sezzle) [reply .npmignore]: done!

File: README.md (hunk 3 above)
Comment (mw-sezzle) [reply README.md]: I've removed this

**Comment by mw-sezzle** [comment]
@alexpanov please review again. I've cleaned up this MR, sorry about the unnecessary additions before.  This installs fine (no need for forcing the npm install) in a repo that contains an internal library that I help maintain at work. Storybook runs and builds correctly with React 18 installed (no use of `--legacy-peer-deps`) on node v16.

**Comment by josejimenez-swipejobs** [comment]
Thank @mw-sezzle for raising this, I also want to update my library to react@18, which will be really helpful. I was checking and the dependency "@mui/styles" only supports react@17 as the new versions of @material/* have to get rid of them (https://mui.com/system/styles/basics/). And te same with "react-inspector", the latest version support react@18 but no te current one.

**Pull request merged**